--- /tmp/jabberd-2.2.17/tools/db-setup.sqlite	2012-02-12 13:38:25.000000000 -0800
+++ ./jabberd2/tools/db-setup.sqlite	2012-08-28 18:49:00.000000000 -0700
@@ -52,6 +52,11 @@ CREATE TABLE "roster-items" (
 
 CREATE INDEX i_rosteri_owner ON "roster-items"("collection-owner");
 
+---
+--- Apple index to speed up jabber_autobuddy operations (still necessary in jabberd 2.2, with i_rosteri_owner added?)
+---
+CREATE INDEX "roster-items-owner-jid-index" on "roster-items" ("collection-owner", "jid");
+
 --
 -- Roster groups
 -- Used by: mod_roster
