--- /tmp/jabberd-2.2.17/sm/sm.h	2012-04-28 10:25:19.000000000 -0700
+++ ./jabberd2/sm/sm.h	2012-08-28 18:49:00.000000000 -0700
@@ -173,6 +173,8 @@ struct sm_st {
     char                *router_pass;       /**< password to authenticate to the router with */
     char                *router_pemfile;    /**< name of file containing a SSL certificate &
                                                  key for channel to the router */
+    char                *router_private_key_password;    /** password for private key if pemfile
+                                                             key is encrypted */
 
     mio_t               mio;                /**< mio context */
 
