--- /tmp/jabberd-2.2.17/router/router.h	2012-05-04 07:26:29.000000000 -0700
+++ ./jabberd2/router/router.h	2012-08-28 18:49:00.000000000 -0700
@@ -93,6 +93,7 @@ struct router_st {
     int                 local_port;
     char                *local_secret;
     char                *local_pemfile;
+    char                *local_private_key_password;
 
     /** max file descriptors */
     int                 io_max_fds;
@@ -159,6 +160,9 @@ struct router_st {
     /** simple message logging */
 	int message_logging_enabled;
 	char *message_logging_file;
+    int message_logging_roll_days;
+    int message_logging_roll_megs;
+    char *filter_muc_messages_from;
 };
 
 /** a single component */
@@ -232,6 +236,7 @@ void    filter_unload(router_t r);
 int     filter_packet(router_t r, nad_t nad);
 
 int     message_log(nad_t nad, router_t r, const unsigned char *msg_from, const unsigned char *msg_to);
+int     roll_message_log(router_t r);
 
 void routes_free(routes_t routes);
 
